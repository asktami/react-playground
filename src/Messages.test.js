import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'; // add snapshot tests

import Messages from './Messages';

describe('Messages component', () => {
	// all my Messages component test cases here

	// smoke test:
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Messages />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	// snapshot testing:
	/*
	add a test case to:
	1.  Render the component and create a human readable JSON file
   	2. Compare the rendered component to a saved version of the component
	*/

	it('renders the UI as expected with unreads > 0', () => {
		const tree = renderer
			.create(<Messages name="messages" unread={4} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	// add a test to perform a snapshot test on the Messages component with the unread prop set to 0
	it('renders the UI as expected with unreads = 0', () => {
		const tree = renderer
			.create(<Messages name="Messages" unread={0} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
