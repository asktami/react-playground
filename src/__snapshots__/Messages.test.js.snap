// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders the UI as expected with unreads = 0 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`Messages component renders the UI as expected with unreads > 0 1`] = `
<div
  className="messages"
>
  messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;
